build/android/BUILD.gn

```diff
@@ -0,0 +1,340 @@
+  # chromium new base. Need to check manually after each rebase.
+  java_source_dirs = [
+    "//base/android/java/src",
+    "//components/web_contents_delegate_android/android/java/src",
+    "//components/navigation_interception/android/java/src",
+    "//content/public/android/java/src",
+    "//media/base/android/java/src",
+    "//net/android/java/src",
+    "//ui/android/java/src",
+    "//xwalk/extensions/android/java/src",
+    "//xwalk/runtime/android/core/src",
+    "//xwalk/runtime/android/core_internal/src",
+    "$root_gen_dir/net/android/net_errors_java__apply_gcc/java_cpp_template/org/chromium/net/NetError.java",
+    "$root_gen_dir/base/base_native_libraries_gen__apply_gcc/java_cpp_template/org/chromium/base/library_loader/NativeLibraries.java",
+    "$root_gen_dir/base/base_multidex_gen__apply_gcc/java_cpp_template/org/chromium/base/multidex/ChromiumMultiDex.java",
+    "$root_gen_dir/resource_map",
+
+    # NativeLibraries.java must be copied later than gen/templates to override the empty NativeLibraries.java in gen/templates.
+    "$root_gen_dir/xwalk/runtime/android/core_internal_empty/xwalk_core_internal_empty_embedder_apk__native_libraries_java__apply_gcc/java_cpp_template/org/chromium/base/library_loader/NativeLibraries.java",
+    "$root_gen_dir/xwalk_core_reflection_layer/bridge",
+    "$root_gen_dir/xwalk_core_reflection_layer/wrapper",
+  ]
+
+  script = "//xwalk/build/android/merge_java_srcs.py"
+  java_source_dirs_list = ""
+  foreach(i, java_source_dirs) {
+    dirsAbsPath = get_path_info(i, "abspath")
+    dirsRebasePath = rebase_path(dirsAbsPath, "$root_build_dir")
+    java_source_dirs_list = "$java_source_dirs_list $dirsRebasePath"
+  }
+  args = [
+    "--dirs=$java_source_dirs_list",
+    "--target-path=./xwalk_core_library_src/src",
+  ]
+  outputs = [
+    "$root_out_dir/xwalk_core_library_src/src",
+  ]
+  inputs = [
+    "$root_out_dir/xwalk_core_library_src/src/README.md",
+  ]
+  deps = [
+    ":generate_resource_map",
+    ":generate_xwalk_core_library_src_package",
+    "//xwalk/runtime/android/core_internal:xwalk_core_reflection_layer_java_gen",
+  ]
+}
+
+action("pack_xwalk_core_library") {
+  script = "//xwalk/tools/tar.py"
+  args = [ "./xwalk_core_library" ]
+  outputs = [
+    "$root_out_dir/xwalk_core_library.tar.gz",
+  ]
+  deps = [
+    ":copy_xwalk_core_library_src",
+    ":xwalk_core_library",
+  ]
+}
+
+action("pack_xwalk_core_library_src") {
+  script = "//xwalk/tools/tar.py"
+  args = [ "./xwalk_core_library_src" ]
+  outputs = [
+    "$root_out_dir/xwalk_core_library_src.tar.gz",
+  ]
+  deps = [
+    ":copy_xwalk_core_library_src",
+    ":xwalk_core_library",
+  ]
+}
+
+action("pack_xwalk_shared_library") {
+  script = "//xwalk/tools/tar.py"
+  args = [ "./xwalk_shared_library" ]
+  outputs = [
+    "$root_out_dir/xwalk_shared_library.tar.gz",
+  ]
+  deps = [
+    ":xwalk_shared_library",
+  ]
+}
+
+action("xwalk_core_library_documentation") {
```

Comment:
> it first packages all the sources into a single source jar and passes that to its javadoc wrapper.
Seems it doesn't package sources into srcjar before passes to javadoc wrapper?  The sources are set by `--lib-java-dir`. So I have to extract it first(like "extract_cronet_jars" action does)
Or is it possible to generate both java files and srcjars in script "reflection_generator.py" with flag "--use-srcjars"? 
Which way do you prefer?